src/Compilers/Core/Portable/PEWriter/Members.cs

```diff
@@ -94,17 +94,19 @@ internal static class CallingConventionUtils
             | SignatureAttributes.Instance
             | SignatureAttributes.ExplicitThis;
 
-        internal static CallingConvention FromSignatureConvention(this SignatureCallingConvention convention, bool throwOnInvalidConvention = false)
+        internal static CallingConvention FromSignatureConvention(this SignatureCallingConvention convention)
         {
-            var callingConvention = (CallingConvention)(convention & SignatureCallingConventionMask);
-            if (throwOnInvalidConvention && callingConvention != (CallingConvention)convention)
+            if (!convention.IsValid())
             {
                 throw new UnsupportedSignatureContent();
             }
 
-            return callingConvention;
+            return (CallingConvention)(convention & SignatureCallingConventionMask);
         }
 
+        internal static bool IsValid(this SignatureCallingConvention convention)
+            => convention <= SignatureCallingConvention.VarArgs || convention == SignatureCallingConvention.Unmanaged;
```

Comment:
>SignatureCallingConvention.VarArgs [](start = 29, length = 34)

VarArgs is a valid calling convention for a regular method. Neither this class, nor this method look specific to function pointers

Reply:
> VarArgs is a valid calling convention for a regular method

Right, and this method considers it to be valid.

> Neither this class, nor this method look specific to function pointers

They're not.

---
In reply to: [463764392](https://github.com/dotnet/roslyn/pull/46467#discussion_r463764392) [](ancestors = 463764392)